src/Orchard.Web/Modules/Orchard.Roles/Conditions/RoleCondition.cs

```diff
@@ -22,15 +22,16 @@ public class RoleConditionProvider : IConditionProvider {
                 return;
             }
 
+            var roles = ((object[])evaluationContext.Arguments).Cast<string>();
             var user = _authenticationService.GetAuthenticatedUser();
-            if (user == null) {
+            if (user == null){
                 evaluationContext.Result = false;
-                return;
+                evaluationContext.Result= roles.Contains("Anonymous");
```

Comment:
Is the format a CSV? Then we should split and check the actual value, otherwise a role name "CustomAnonymous" would still work